src/engine.ts

```diff
@@ -644,6 +701,70 @@ class StyleElementController extends NodeController<HTMLStyleElement> {
   }
 }
 
+class ShadowRootController extends NodeController<ShadowRoot> {
+  private controller: AbortController | null = null;
+  private mo: MutationObserver | null = null;
+  private host: HTMLElement | null = null;
```

Comment:
These don't seem like they need the `null` case?

Reply:
will get this updated